test/function-test.js

```diff
@@ -33,3 +33,55 @@ describe('apply', () => {
     assert.strictEqual(apply(f, x), f(x));
   });
 });
+
+describe('curry', () => {
+  describe('curry2', () => {
```

Comment:
It's interesting to think about what the invariants of curry are.  Accounting for varargs, I believe the invariant of curry2 is:

for all f, a, b:  `curry2(f)(a)(b) === curry2(f)(a, b) === f(a, b)`

Similarly, for curry3:

for all f, a, b, c: `curry3(f)(a)(b)(c) === curry3(f)(a, b)(c) === curry3(f)(a)(b, c) === curry3(f)(a, b, c) === f(a, b, c)`

How about adding tests to verify those invariants?